File: src/main/java/com/github/ccguyka/showupdates/objects/DependencyUpdates.java
package com.github.ccguyka.showupdates.objects;

import java.util.List;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.google.common.base.MoreObjects;
import java.util.Objects;

public class DependencyUpdates {

    private final List<ArtifactUpdate> artifacts;

    @JsonCreator
    public DependencyUpdates(@JsonProperty("artifacts") List<ArtifactUpdate> artifacts) {
        this.artifacts = artifacts;
    }

    public List<ArtifactUpdate> getArtifacts() {
        return artifacts;
    }

    @Override
    public String toString() {
        return MoreObjects.toStringHelper(this).add("artifacts", artifacts).toString();
    }

    @Override
    public int hashCode() {
        return Objects.hashCode(artifacts);
    }

    @Override
    public boolean equals(Object object) {
        if (object instanceof DependencyUpdates) {
            DependencyUpdates that = (DependencyUpdates) object;
            return Objects.equals(this.artifacts, that.artifacts);
        }
        return false;
    }
}

File: src/main/kotlin/com/github/ccguyka/showupdates/objects/DependencyUpdates.kt
package com.github.ccguyka.showupdates.objects

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty
import com.google.common.base.MoreObjects
import java.util.*

class DependencyUpdates @JsonCreator constructor(@param:JsonProperty("artifacts") val artifacts: List<ArtifactUpdate>) {

    override fun toString(): String {
        return MoreObjects.toStringHelper(this).add("artifacts", artifacts).toString()
    }

    override fun hashCode(): Int {
        return Objects.hashCode(artifacts)
    }

    override fun equals(other: Any?): Boolean {
        if (other is DependencyUpdates) {
            return artifacts == other.artifacts
        }
        return false
    }
}
